src/main/org/codehaus/groovy/classgen/asm/OperandStack.java

```diff
@@ -336,14 +336,12 @@ private void doConvertAndCast(ClassNode targetType, boolean coerce) {
                 return;
             }
             box();
-        } else if (primTop) {
-            // top is primitive, target is not
-            // so box and do groovy cast
-            controller.getInvocationWriter().castToNonPrimitiveIfNecessary(top, targetType);
         } else if (primTarget) {
             // top is not primitive so unbox
             // leave that BH#doCast later
         } else {
+            // top is primitive, target is not
+            // so box and do groovy cast
```

Comment:
the comment needs a change... "// top might be primitive, target is not" and "// so let invocation writer box if needed and do groovy cast otherwise". I think this case appeared to due to moving some functionality into the invocation writer, I think before there has been really two different actions. Well.. Still can make this change with the changed comment then of course ;)